src/main/java/botimusprime/tasks/TaskList.java

```diff
@@ -0,0 +1,277 @@
+
+        int idx = Integer.parseInt(numFinder[1]) - 1;
+        Task task = tasks.get(idx);
+        tasks.remove(idx);
+        storage.saveToDisk(this);
+        System.out.println("____________________________________________________________\n"
+                + "Noted. I've removed this task:\n" + task + "\nNow you have "
+                        + tasks.size() + " tasks in the list.");
+    }
+
+    /**
+     * Adds a new To\Do task to the list based on the input.
+     * If the input does not contain a valid description, it displays an error message.
+     *
+     * @param input The user input containing the full command.
+     */
+    public void addToDo(String input) {
+        if (input.length() <= 5) {
+            System.out.println("eh bro udw to put ur description of ur task issit");
+            return;
+        }
+
+        ToDo task = new ToDo(input.substring(5), false);
+
+        tasks.add(task);
+        storage.saveToDisk(this);
+
+        System.out.println(
+                "____________________________________________________________\n" +
+                        String.format("Alright, I've added the task:\n %s\nNow you have %d tasks in the list.\n",
+                                task, tasks.size())
+                                        + "____________________________________________________________\n");
+    }
+
+    /**
+     * Adds a new Deadline task to the list based on the input.
+     * If the input does not contain a valid description or deadline, it displays an error message.
+     *
+     * @param input The user input containing the full command.
+     */
+    public void addDeadline(String input) {
+        if (input.length() <= 9 || !input.contains("/by")) {
+            System.out.println("brother u forgot to type all the deadline task details plz.");
+        }
+        String[] parser = input.split("/by ");
+
+        if (parser.length < 2 || parser[1].trim().isEmpty()) {
+            System.out.println("wheres the deadline!!");
+            return;
+        }
+        String description = parser[0].substring(9);
+        String deadline = parser[1];
+
+        if (deadline.isEmpty()) {
+            System.out.println("eh bro u got due date anot");
+            return;
+        } else if (description.isEmpty()) {
+            System.out.println("eh bro ur task no description leh wake up ur idea");
+        }
+
+        Deadline task = new Deadline(description,
+                false,
+                        Parser.stringToDateTime(deadline));
+
+        tasks.add(task);
+        storage.saveToDisk(this);
+
+        System.out.println(
+                "____________________________________________________________\n" +
+                        String.format("Alright, I've added the task:\n %s\nNow you have %d tasks in the list.\n",
+                                task, tasks.size())
+                                        + "____________________________________________________________\n");
+    }
```

Comment:
You could add method that returns Strinf for the new line to reduce the code length.